dshowcapture.hpp

```diff
@@ -202,6 +203,9 @@ struct AudioConfig : Config {
 
 class DSHOWCAPTURE_EXPORT Device {
 	HDevice *context;
+	DeviceDialogBox *videoDialog;
+	DeviceDialogBox *crossbarDialog;
+	DeviceDialogBox *audioDialog;
```

Comment:
Not very familiar with this code, what's the background on having these new `videoDialog`, `crossbarDialog` and `audioDialog`?

Reply:
You can have a video dialog box that is for the camera, an audio box for audio settings of a device and also a crossbar dialog box. If a device have those 3 settings we need to open 3 different dialog boxes when a user click their respective buttons. DeviceDialogBox handles it's own dialog box thread. That is why 3 dialog boxes.
![image](https://user-images.githubusercontent.com/8325447/80826006-5b386800-8b96-11ea-8447-3581fbadeba2.png)

Reply:
I see, got it.